### PR TITLE
Tea fix

### PR DESCRIPTION
**Pull request by stoeffel**
`htmlToString` returns an empty string if the view contains a `Html.App.map`.
This change to the decoder fixes this.
Let me know if you want me to change anything.

### CODE REVIEW

**Comment by eeue56** [comment]
yep, seems right. please tag and bump in the main repo

File: src/ServerSide/InternalTypes.elm [hunk 1]
```diff
@@ -75,6 +75,9 @@ decodeNodeType =
                     "node" ->
                         Json.Decode.map NodeEntry (decodeNode)
 
+                    "tagger" ->
+                        Json.Decode.map NodeEntry (decodeTagger)
```
Comment (eeue56) [inline src/ServerSide/InternalTypes.elm]: this might need to be recursive

Comment (stoeffel) [reply src/ServerSide/InternalTypes.elm]: will add a test. good catch

Comment (stoeffel) [reply src/ServerSide/InternalTypes.elm]: `decodeTagger` uses `decodeNode` which is recursive. I added a test and it works.

File: src/ServerSide/InternalTypes.elm [hunk 2]
```diff
@@ -93,6 +96,11 @@ encodeTextTag { text } =
     Json.Encode.object [ ( "text", Json.Encode.string text ) ]
 
 
+decodeTagger : Json.Decode.Decoder NodeRecord
+decodeTagger =
+    Json.Decode.at [ "node" ] decodeNode
```
Comment (eeue56) [inline src/ServerSide/InternalTypes.elm]: I think you'll need to test `Html.App.map Foo (text "")` too

Comment (eeue56) [reply src/ServerSide/InternalTypes.elm]: and `Html.App.map Foo (Html.App.map Bar (div [] []))`

File: src/ServerSide/InternalTypes.elm [hunk 3]
```diff
@@ -93,6 +96,15 @@ encodeTextTag { text } =
     Json.Encode.object [ ( "text", Json.Encode.string text ) ]
 
 
+decodeTagger : Json.Decode.Decoder NodeType
+decodeTagger =
+    Json.Decode.oneOf
+        [ Json.Decode.at [ "node" ] decodeNodeType
+        , Json.Decode.at [ "text" ] decodeNodeType
+        , Json.Decode.at [ "custom" ] decodeNodeType
```
Comment (stoeffel) [inline src/ServerSide/InternalTypes.elm:104]: @eeue56 I think this should work in all cases.

Comment (eeue56) [reply src/ServerSide/InternalTypes.elm:104]: :+1:

**Pull request merged**